Marlin/src/feature/bedlevel/ubl/ubl_motion.cpp

```diff
@@ -85,8 +85,8 @@
       #endif
 
       // The distance is always MESH_X_DIST so multiply by the constant reciprocal.
-      const float xratio = (end.x - get_mesh_x(iend.x)) * RECIPROCAL(MESH_X_DIST),
-                  yratio = (end.y - get_mesh_y(iend.y)) * RECIPROCAL(MESH_Y_DIST),
+      const float xratio = (end.x - get_mesh_x(iend.x)) * RECIPROCAL(get_mesh_x_dist()),
+                  yratio = (end.y - get_mesh_y(iend.y)) * RECIPROCAL(get_mesh_y_dist()),
```

Comment:
The use of the reciprocal with a constant here is to reduce the computational cost, replacing a float division on every segment with a float multiply. To retain this optimization the firmware should pre-calculate and cache these reciprocals whenever the probing region or number of X / Y points changes.